packages/cli/src/cms/api/api-client.ts

```diff
@@ -110,46 +148,53 @@ class RemotesApiClient {
     }
 
     payload.isMainBranch && formData.append('isMainBranch', 'true');
-
-    const response = await fetch(
-      `${this.domain}/api/orgs/${organizationId}/projects/${projectId}/pushes`,
-      {
-        method: 'POST',
-        headers: {
-          Authorization: `Bearer ${this.apiKey}`,
-        },
-        body: formData,
-        agent: getProxyAgent(),
-      }
-    );
-
     try {
+      const response = await this.request(
+        `${this.domain}/api/orgs/${organizationId}/projects/${projectId}/pushes`,
+        {
+          method: 'POST',
+          headers: {
+            Authorization: `Bearer ${this.apiKey}`,
+          },
+          body: formData,
+        }
+      );
+
       return await this.getParsedResponse<PushResponse>(response);
     } catch (err) {
-      throw new Error(`Failed to push: ${err.message || 'Unknown error'}`);
-    }
-  }
+      const message = `Failed to push. ${err.message || 'Unknown error'}`;
 
-  async getRemotesList(organizationId: string, projectId: string, mountPath: string) {
-    const response = await fetchWithTimeout(
-      `${this.domain}/api/orgs/${organizationId}/projects/${projectId}/remotes?filter=mountPath:/${mountPath}/`,
-      {
-        method: 'GET',
-        headers: {
-          'Content-Type': 'application/json',
-          Authorization: `Bearer ${this.apiKey}`,
-        },
+      if (err instanceof ReuniteApiError) {
+        throw new ReuniteApiError(message, err.status);
       }
-    );
 
-    if (!response) {
-      throw new Error(`Failed to get remotes list.`);
+      throw new Error(message);
     }
+  }
 
+  async getRemotesList(organizationId: string, projectId: string, mountPath: string) {
```

Comment:
I'd use object as an argument here for better readability: 

```suggestion
  async getRemotesList({organizationId, projectId, mountPath}) {
```